guides/computation-development.md

```diff
@@ -0,0 +1,411 @@
+    type: 'function',
+    fn: function(params) {
+      // ...
+    },
+  },
+  // ...
+};
+```
+
+Authors may add more steps by employing a JavaScript array as the `local` or `remote` value:
+
+```js
+module.exports = {
+  // ...
+
+  local: [
+    // First computation:
+    {
+      type: 'function',
+      fn: function(params) {
+        // ...
+      },
+    },
+
+    // Second computation:
+    {
+      type: 'function',
+      fn: function(params) {
+        // ...
+      },
+    },
+
+    // ...and so on.
+  ],
+  remote: [
+    {
+      type: 'function',
+      fn: function(params) {
+        // ...
+      },
+    },
+    {
+      type: 'function',
+      fn: function(params) {
+        // ...
+      },
+    },
+  ],
+};
+```
+
+### Plugins
+
+Plugins modify your decentralized computation by adjusting its COINSTAC lifecycle or adding functionality. You can add them as a JavaScript array under the `plugins` property:
+
+```js
+module.exports = {
+  name: 'my-computation',
+  version: '1.0.0',
+  local: [
+    // ...
+  ],
+  remote: [
+    // ...
+  ],
+  plugins: ['group-step'],
+};
+```
+
+The only plugin currently available is *group-step*. It is a flow control modifier that ensures that all clients have completed a computation “step” before passing control back to the server.
+
+## Testing Computations
+
+To test your computation you’ll need to provide some basic parameters and pass things to _[coinstac-simulator](https://github.com/MRN-Code/coinstac/tree/master/packages/coinstac-simulator)_. It needs a declaration file, which is written in JavaScript:
+
+```js
+// declaration.js
+module.exports = {
+  computationPath: './path/to/computation.js',
+  local: [
+    {
+      x: 1,
+      y: 2,
+    }, {
+      x: 3,
+      y: 4,
+    }
+  ],
+  remote: {
+    x: 5,
+    y: 6,
+  },
+};
+```
+
+This declaration has a few properties:
+
+* **`computationPath`** `<String>`: Path to your decentralized computation file.
+* **`local`** `<Array>`: Collection of local data to simulate. Each object in this array equates to a client instance, and the client is passed the data in the object.
+* **`remote`** `<Object>`: (optional) data for the server.
+
+```shell
+coinstac-simulator --declaration ./path/to/declaration.js
+```
+
+### File-loading Utilities
+
+_coinstac-simulator_ comes with two utility methods to assist in loading your data. To use them you’ll need to set up a module with a [_package.json_ via NPM](https://docs.npmjs.com/getting-started/using-a-package.json):
+
+1. Locate or create a directory suitable for your algorithm testing.
+2. Run `npm init` and follow the prompts. This will generate a _package.json_ file, which NPM uses to manage dependencies.
+3. Run `npm install coinstac-simulator --save` to add _coinstac-simulator_ as a dependency to your project. This will create a _node_modules_ directory – where NPM places packages – in your project directory.
+
+Here’s the API:
+
+#### coinstacSimulator.loadFiles(pattern, [delimiter]):
```

Comment:
I think this section needs a bit more info to be useful, I can do it if you'd like